worlds/faxanadu/Rules.py

```diff
@@ -0,0 +1,79 @@
+    set_rule(multiworld.get_location("Eolis Red Potion", player), lambda state: can_buy_in_eolis(state, player))
+    set_rule(multiworld.get_location("Path to Apolune Magic Shield", player), lambda state: state.has("Key King", player))  # Mid-late cost, make sure we've progressed
+    set_rule(multiworld.get_location("Path to Apolune Death", player), lambda state: state.has("Key Ace", player))  # Mid-late cost, make sure we've progressed
+    set_rule(multiworld.get_location("Tower of Trunk Hidden Mattock", player), lambda state:
+             # This is actually possible if the monster drop into the stairs and kill it with dagger. But it's a "pro move"
+             state.has("Deluge", player, 1) or
+             state.has("Progressive Sword", player, 2))
+    set_rule(multiworld.get_location("Path to Forepaw Glove", player), lambda state:
+             state.has_all(["Deluge", "Unlock Wingboots"], player))
+    set_rule(multiworld.get_location("Trunk Red Potion", player), lambda state: state.has("Unlock Wingboots", player))
+    set_rule(multiworld.get_location("Sky Spring", player), lambda state: state.has("Unlock Wingboots", player))
+    set_rule(multiworld.get_location("Tower of Fortress Spring", player), lambda state: state.has("Spring Elixir", player))
+    set_rule(multiworld.get_location("Tower of Fortress Guru", player), lambda state: state.has("Sky Spring Flow", player))
+    set_rule(multiworld.get_location("Tower of Suffer Hidden Wingboots", player), lambda state: 
+             state.has("Deluge", player) or
+             state.has("Progressive Sword", player, 2))
+    set_rule(multiworld.get_location("Misty House", player), lambda state: state.has("Black Onyx", player))
+    set_rule(multiworld.get_location("Misty Doctor Office", player), lambda state: has_any_magic(state, player))
+    set_rule(multiworld.get_location("Conflate Guru", player), lambda state: state.has("Progressive Armor", player, 3))
+    set_rule(multiworld.get_location("Branches Hidden Mattock", player), lambda state: state.has("Unlock Wingboots", player))
+    set_rule(multiworld.get_location("Path to Daybreak Glove", player), lambda state: state.has("Unlock Wingboots", player))
+    set_rule(multiworld.get_location("Dartmoor Castle Hidden Hourglass", player), lambda state: state.has("Unlock Wingboots", player))
+    set_rule(multiworld.get_location("Dartmoor Castle Hidden Red Potion", player), lambda state: has_any_magic(state, player))
+    set_rule(multiworld.get_location("Fraternal Castle Guru", player), lambda state: state.has("Progressive Sword", player, 4))
+    set_rule(multiworld.get_location("Fraternal Castle Shop Hidden Ointment", player), lambda state: has_any_magic(state, player))
+
+    if faxanadu_world.options.require_dragon_slayer.value:
+        set_rule(multiworld.get_location("Evil One", player), lambda state: 
+                state.has_all_counts({"Progressive Sword": 4, "Progressive Armor": 3, "Progressive Shield": 4}, player))
```

Comment:
One space off alignment
```suggestion
                 state.has_all_counts({"Progressive Sword": 4, "Progressive Armor": 3, "Progressive Shield": 4}, player))
```